components/common/src/command/package/install.rs

```diff
@@ -702,11 +705,35 @@ impl<'a> InstallTask<'a> {
     where
         T: UIWriter,
     {
-        artifact.unpack(Some(self.fs_root_path))?;
-        ui.status(Status::Installed, artifact.ident()?)?;
+        let ident = &artifact.ident()?;
+        let real_install_path = &pkg_install_path(ident, Some(self.fs_root_path));
+        let real_install_base = real_install_path.parent().ok_or(
+            Error::PackageUnpackFailed(
+                "Could not determine parent directory for package unpack location".to_owned()
+            )
+        )?;
+        let temp_dir = self.temporary_unpack_directory(real_install_base, real_install_path)?;
+        let temp_install_path = &pkg_install_path(ident, Some(temp_dir.path()));
+        artifact.unpack(Some(temp_dir.path()))?;
+        fs::rename(temp_install_path, real_install_path)?;
+        fs::File::open(real_install_base).and_then(|f| f.sync_all())?;
```

Comment:
Looks like this is the culprit and I think it is unnecessary. The error is likely occuring because `real_install_base` is a directory. I don't think a `sync` is necessary after a rename. Only after a `write`.

Reply:
I think Sync is necessary on some unix file systems to ensure the rename operation is commited to disk.